includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -236,4 +237,43 @@ private function is_gif_url( $url ) {
 		$path = AMP_WP_Utils::parse_url( $url, PHP_URL_PATH );
 		return substr( $path, -strlen( $ext ) ) === $ext;
 	}
+
+	/**
+	 * Handles an issue with the aligncenter class.
+	 *
+	 * If the <amp-img> has the class aligncenter, this strips the class and wraps it in a <figure> to center the image.
+	 * There was an issue where the aligncenter class overrode the "display: inline-block" rule of AMP's layout="intrinsic" attribute.
+	 * So this strips that class, and instead wraps the image in a <figure> to center it.
+	 *
+	 * @since 0.7
+	 * @see https://github.com/Automattic/amp-wp/issues/1104
+	 *
+	 * @param DOMNode|DOMElement $node The <amp-img> node.
+	 * @return DOMNode|DOMElement $node The <amp-img> node, possibly wrapped in a <figure>.
+	 */
+	public function handle_centering( $node ) {
+		$align_class = 'aligncenter';
+		$classes     = $node->getAttribute( 'class' );
+		$width       = $node->getAttribute( 'width' );
+
+		// If this doesn't have a width attribute, centering it in the <figure> wrapper won't work.
+		if ( ( false === strpos( $classes, $align_class ) ) || empty( $width ) ) {
+			return $node;
+		}
+
+		// Strip the class, and wrap the <amp-img> in a <figure>.
+		$classes = trim( str_replace( $align_class, '', $classes ) );
```

Comment:
We could probably also use this logic to center images in legacy templating if they don't have an `alignleft` or `alignright` class. In `0.6`, they were centered in those cases ([more details](https://github.com/Automattic/amp-wp/issues/1104#issuecomment-385586532)). We could use this to detect "legacy templating":

```php
! current_theme_supports( 'amp' )
```

Reply:
If we did that, then I'd prefer to pass that flag as an argument to the class as a whole instead of calling this global function. For example, there could be an arg like `auto_align_center` passed in:

https://github.com/Automattic/amp-wp/blob/cd03ee45171e41b3c7ae111afcda58584f96a523/includes/amp-helper-functions.php#L482

If `! current_theme_supports( 'amp' )`.

Reply:
Thanks, @westonruter. 

@mehigh and I discussed how the legacy templating in `0.6` centers images that don't have `alignleft` and `alignright`, and maybe we don't need to continue that behavior in `0.7`. 

At least images with `alignnone` shouldn't be centered. They're centered in `0.6 ` ([example](https://ryankienstra.wordpress.com/2018/05/01/test-page/amp/)), but they're not centered in `0.7`.